src/utils.ts

```diff
@@ -155,11 +162,28 @@ const fetchWithHeaders = async (
   } as HeadersInit
 
   // Normalizes headers into a plain object format.
-  options.headers = normalizeHeaders(options.headers);
-  
+  options.headers = normalizeHeaders(options.headers)
+
+  // Filter custom headers for ollama.com and inject Authorization from environment 
+  const hostname = new URL(url).hostname
+  if (hostname === 'ollama.com') {
+    // For ollama.com, only keep Authorization header
+    const authEntry = Object.entries(options.headers).find(([k]) => k.toLowerCase() === 'authorization')
+    options.headers = authEntry ? { [authEntry[0]]: authEntry[1] } : {}
+
+    // Inject Authorization from environment if not already provided 
+    const hasAuth = authEntry !== undefined
+    const apiKey = typeof process !== 'undefined' && process.env?.OLLAMA_API_KEY
+    if (!hasAuth && apiKey) {
+      options.headers['authorization'] = `Bearer ${apiKey}`
```

Comment:
```suggestion
      options.headers['Authorization'] = `Bearer ${apiKey}`
```

I think the proper usage is capitalized?

Reply:
i'm not sure actually. i think technically yes but in the python repo we were standardizing on lowercase. we can pick whatever is more correct and i can update on both